Test for newlines at end of all *.rs and *.toml files

mbestavros opened the pull request:
Resolves #160.

mbestavros commented:
I've rewritten the test to better follow the format we've established for our other tests. It also correctly tests all files now. Ready for review.

npmccallum commented on hooks/pre-push.d/eof-newline-test:
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+top=`git rev-parse --show-toplevel`
+status=0
+for file in $(find . -path ./target -prune -o -type f -name "*.rs" -o -name "*.toml"); do
```
```suggestion
for file in $(find . -path ./target -prune -o -type f -name "*.rs" -o -type f -name "*.toml"); do
```

I think we need `-type f` again after the or.

mbestavros replied:
Good catch. Fixed in latest push.

npmccallum commented on hooks/pre-push.d/eof-newline-test:
```diff
@@ -0,0 +1,16 @@
+top=`git rev-parse --show-toplevel`
+status=0
+for file in $(find . -path ./target -prune -o -type f -name "*.rs" -o -name "*.toml"); do
+    if [ -f "$file" ] && [ ! -z $(tail -c 1 "$file") ]; then
```
```suggestion
    if [ ! -z $(tail -c 1 "$file") ]; then
```

Also, this tests for any whitespace, not specifically newline.

mbestavros replied:
Right. Fixed in latest push.

npmccallum commented on hooks/pre-push.d/eof-newline-test:
```diff
@@ -0,0 +1,16 @@
+
+#  -print0 | xargs -0 -L1 bash -c 'test "$(tail -c 1 "$0")" && echo "$0"')
+# [ -z "$files_without_eof" ] && exit 0
+# printf "The following files are missing a newline at EOF:\n$files_without_eof\n" && exit 1
```
Clean this up.

mbestavros replied:
D'oh, sorry. Fixed in latest push.

mbestavros commented:
I've addressed the first round of feedback. I've also incorporated `-print` in the `find` invocation to suppress printing the `target` directory, as suggested by @npmccallum in #162.
